### DISCUSSION TITLE
Python Import Error : Unable to to import JWTBearerToken

### DISCUSSION THREAD

**Discussion by dcs3spp** [lepture/authlib#345] [Q&A] [answered]
Hi,

I am trying to use ```JWTBearerToken``` from ```authlib.oauth2.rfc7523```.  However, I have encountered import error when trying to import ```JWTBearerToken```:
 
**Create a virtual environment and install authlib**
```console
$ pyenv virtualenv authlib
$ pyenv activate authlib
$ python -m pip install --upgrade pip
$ pip install authlib
```

**Start python and try to import ```JWTBearerToken```**
```python
>>> from authlib.oauth2.rfc7523 import JWTBearerToken
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ImportError: cannot import name 'JWTBearerToken' from 'authlib.oauth2.rfc7523' (/Users/simon/.pyenv/versions/authlib/lib/python3.8/site-packages/authlib/oauth2/rfc7523/__init__.py)
>>> 
```

**Start python and try and import via validator module**
```python
>>> from authlib.oauth2.rfc7523.validator import JWTBearerToken 
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ModuleNotFoundError: No module named 'authlib.oauth2.rfc7523.validator'
```

The source for the ```authlib.oauth2.rfc7523``` module is [here](https://github.com/lepture/authlib/blob/169c7dcfc47478c8d55553cc95fb0f5578162b77/authlib/oauth2/rfc7523/__init__.py).

What is the correct way to import JWTBearerToken?

**Comment by lepture** [comment, marked as answer]
There must be something wrong with your environment. First, you need to find which version is your authlib?

```py
>>> import authlib
>>> print(authlib.__version__)
```

**Reply by dcs3spp** [reply]
Thanks @lepture. 

```python
import authlib
print(authlib.__version__)
```

Gives version *0.15.3*

The virtual environment python version is 3.8.2. It was created using the steps detailed above, via pyenv.

**Reply by dcs3spp** [reply]
Hi @lepture,

I have tried upgrading to Python 3.9.5 on a macos Catalina and Ubuntu Linux environment via pyenv.

In both cases I have followed this procedure:
```console
$ pyenv virtualenv 3.9.5 authlib
$ pyenv activate authlib
$ python -m pip install --upgrade pip
$ pip install --no-cache authlib
```

In both cases :

```python
import authlib
print(authlib.__version__)
```

Prints 0.15.3.

In both cases I am unable to install ```JWTBearerToken``` from ```authlib.oauth2.rfc7523```:

```bash
>>> from authlib.oauth2.rfc7523 import JWTBearerToken
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ImportError: cannot import name 'JWTBearerToken' from 'authlib.oauth2.rfc7523' >>> 
```

Would it be possible for someone to confirm that they are able to get this import working in a virtual environment on their macos or Linux platform?

**Reply by lepture** [reply]
@dcs3spp you are reading the wrong documentation. 

```
>>> from authlib.oauth2.rfc7523 import JWTBearerToken
```

This must be code in Authlib 1.0.0. There is no `JWTBearerToken` in 0.15.x. You can check the 0.15 code at https://github.com/lepture/authlib/tree/maintain-0.15/authlib/oauth2/rfc7523
